specs/deneb/polynomial-commitments-sampling.md

```diff
@@ -0,0 +1,530 @@
+# Deneb -- Polynomial Commitments
+
+## Table of contents
+
+<!-- TOC -->
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+- [Introduction](#introduction)
+- [Custom types](#custom-types)
+- [Constants](#constants)
+- [Preset](#preset)
+  - [Samples](#samples)
+  - [Crypto](#crypto)
+- [Helper functions](#helper-functions)
+  - [Linear combinations](#linear-combinations)
+    - [`g2_lincomb`](#g2_lincomb)
+  - [FFTs](#ffts)
+    - [`_simple_ft_field`](#_simple_ft_field)
+    - [`_fft_field`](#_fft_field)
+    - [`fft_field`](#fft_field)
+  - [Polynomials in coefficient form](#polynomials-in-coefficient-form)
+    - [`polynomial_eval_to_coeff`](#polynomial_eval_to_coeff)
+    - [`polynomial_coeff_to_eval`](#polynomial_coeff_to_eval)
+    - [`add_polynomialcoeff`](#add_polynomialcoeff)
+    - [`neg_polynomialcoeff`](#neg_polynomialcoeff)
+    - [`multiply_polynomialcoeff`](#multiply_polynomialcoeff)
+    - [`divide_polynomialcoeff`](#divide_polynomialcoeff)
+    - [`shift_polynomialcoeff`](#shift_polynomialcoeff)
+    - [`interpolate_polynomialcoeff`](#interpolate_polynomialcoeff)
+    - [`zero_polynomialcoeff`](#zero_polynomialcoeff)
+    - [`evaluate_polynomialcoeff`](#evaluate_polynomialcoeff)
+  - [KZG multiproofs](#kzg-multiproofs)
+    - [`compute_kzg_proof_multi_impl`](#compute_kzg_proof_multi_impl)
+    - [`verify_kzg_proof_multi_impl`](#verify_kzg_proof_multi_impl)
+  - [Sample cosets](#sample-cosets)
+    - [`coset_for_sample`](#coset_for_sample)
+- [Samples](#samples-1)
+  - [Sample computation](#sample-computation)
+    - [`compute_samples_and_proofs`](#compute_samples_and_proofs)
+    - [`compute_samples`](#compute_samples)
+  - [Sample verification](#sample-verification)
+    - [`verify_sample_proof`](#verify_sample_proof)
+    - [`verify_sample_proof_batch`](#verify_sample_proof_batch)
+- [Reconstruction](#reconstruction)
+  - [`recover_samples_impl`](#recover_samples_impl)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+<!-- /TOC -->
+
+## Introduction
+
+This document extends [polynomial-commitments.md](polynomial-commitments.md) with the functions required for data availability sampling (DAS). It is not part of the core Deneb spec but an extension that can be optionally implemented to allow nodes to reduce their load using DAS.
+
+For any KZG library extended to support DAS, functions flagged as "Public method" MUST be provided by the underlying KZG library as public functions. All other functions are private functions used internally by the KZG library.
+
+Public functions MUST accept raw bytes as input and perform the required cryptographic normalization before invoking any internal functions.
+
+## Custom types
+
+| Name | SSZ equivalent | Description |
+| - | - | - |
+| `PolynomialCoeff` | `Vector[BLSFieldElement, FIELD_ELEMENTS_PER_BLOB]` | A polynomial in coefficient form |
```

Comment:
It seems like `Polynomial` is used but not defined here.
Maybe we should use `PolynomialEval`?

Reply:
It's a possibility, but it would also require changing the existing cryptography specifications (and corresponding libraries). @jtraglia?

Reply:
Sorry somehow I missed this. Yeah it's possible but not worth it IMO.